v2/oauth.go

```diff
@@ -0,0 +1,40 @@
+package tsclient
+
+import (
+	"context"
+	"net/http"
+	"path"
+
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// OAuthConfig provides a mechanism for configuring OAuth authentication.
+type OAuthConfig struct {
+	// ClientID is the client ID of the OAuth client.
+	ClientID string
+	// ClientSecret is the client secret of the OAuth client.
+	ClientSecret string
+	// Scopes are the scopes to request when generating tokens for this OAuth client.
+	Scopes []string
+	// BaseURL is an optional base URL for the API server to which we'll connect. Defaults to https://api.tailscale.com.
+	BaseURL string
+}
+
+// HTTPClient constructs an HTTP client that authenticates using OAuth.
+func (ocfg OAuthConfig) HTTPClient() *http.Client {
```

Comment:
maybe just name this Client instead of HTTPClient? Like clientcredentials.Config calls it on line 37 anyway.

Looking at the example usage in docs, there's a lot of "HTTP" around there that's redundant.

Reply:
Yeah, I considered that, but we already have the `Client` type and I wanted to differentiate from that.

Reply:
sgtm